blocks/i-lens/i-lens.js

```diff
@@ -114,26 +87,77 @@ function addStocksData(ilensBody, dropdown = 'default') {
   });
 }
 
-function addSecondDropDown(dropdownsDiv, dropdownValue) {
-  const ilensBody = dropdownsDiv.closest('.i-lens-container').querySelector('.i-lens-body');
+function createDropdown(ilensContainer, menuItems, dropDownDetails = []) {
+  // console.log(dropdownValue);
+  // const menuItems = dropdownValue.split(', ');
+  const dropdownText = menuItems[0].text;
+
+  const dropdownSelectDiv = document.createElement('div');
+  dropdownSelectDiv.className = 'dropdown-select';
+
+  const button = document.createElement('button');
+  button.className = 'dropdown-toggle';
+  button.innerHTML = `<span class="dropdown-text">${dropdownText}</span><span class="icon-down-arrow icon"></span>`;
+
+  const dropdownMenuContainer = document.createElement('div');
+  dropdownMenuContainer.className = 'dropdown-menu-container';
+
+  const ul = document.createElement('ul');
+  ul.className = 'dropdown-menu';
+
+  menuItems.forEach((item) => {
+    const li = document.createElement('li');
+    if (item.value) {
+      li.dataset.value = item.value;
+    }
+    const a = document.createElement('a');
+    const span = document.createElement('span');
+    span.textContent = item.text;
+    a.appendChild(span);
+    li.appendChild(a);
+    li.addEventListener('click', (event) => {
+      // eslint-disable-next-line no-use-before-define
```

Comment:
can get rid of this by placing `updateRecommedations ` above from this.

Reply:
it is sort of creating a cycle, have to keep this comment at one of the method